cpp/cmake/BuildUtils.cmake

```diff
@@ -27,7 +61,7 @@ function(add_gandiva_unit_test REL_TEST_NAME)
     ${CMAKE_SOURCE_DIR}/src
   )
   target_link_libraries(${TEST_NAME}
-    PRIVATE ${ARROW_LIB} gtest_main Boost::boost
+    PRIVATE ${ARROW_LIB_SHARED} gtest_main Boost::boost
```

Comment:
also LLVM::LLVM_SHARED ? or, add this only the test has "llvm" in the name.

Reply:
yup the same is happening already in line 56..i am unable to replace to that with a target..